### PR TITLE
pythonPackages: set mainProgram to pname by default

### PR DESCRIPTION
**Pull request by teto**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
What is this for? What should it achieve? Please write a commit message.

**Comment by teto** [comment]
You are right, I've updated the commit (with a message too)

**Review by FRidh — APPROVED** [review]

File: pkgs/development/interpreters/python/mk-python-derivation.nix [hunk 1]
```diff
@@ -176,6 +176,8 @@ let
       # default to python's platforms
       platforms = python.meta.platforms;
       isBuildPythonPackage = python.meta.platforms;
+    } // lib.optionalAttrs (attrs?pname) {
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/python/mk-python-derivation.nix:179]:
```suggestion
    } // lib.optionalAttrs (attrs ? pname) {
```

Comment (teto) [reply pkgs/development/interpreters/python/mk-python-derivation.nix:179]: I prefered to keep it consistent with the rest.

**Pull request merged**